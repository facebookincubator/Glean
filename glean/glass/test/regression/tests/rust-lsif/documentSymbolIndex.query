action: documentSymbolIndex
args: {"include_refs":true,"filepath":"src/lib.rs","repository":"test"}

