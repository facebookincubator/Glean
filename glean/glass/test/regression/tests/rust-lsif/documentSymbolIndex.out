[
    "@generated",
    {
        "referenced_file_digests": {},
        "revision": "testhash",
        "size": 31,
        "symbols": {
            "12": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 15,
                        "columnEnd": 16,
                        "lineBegin": 12,
                        "lineEnd": 12
                    },
                    "sym": "test/rs/lsif/basics::n"
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::sum_f1"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nbasics\n```\n\n```rust\npub fn sum_f1(n: u64) -> u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 8,
                        "columnEnd": 14,
                        "lineBegin": 12,
                        "lineEnd": 12
                    },
                    "sym": "test/rs/lsif/basics::sum_f1"
                }
            ],
            "13": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::f1_go"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nbasics\n```\n\n```rust\nfn f1_go(n: u64, acc: u64) -> u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 10,
                        "lineBegin": 13,
                        "lineEnd": 13
                    },
                    "sym": "test/rs/lsif/basics::f1_go",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 4,
                            "columnEnd": 9,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 11,
                        "columnEnd": 12,
                        "lineBegin": 13,
                        "lineEnd": 13
                    },
                    "sym": "test/rs/lsif/basics::n",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 15,
                            "columnEnd": 16,
                            "lineBegin": 12,
                            "lineEnd": 12
                        },
                        "repository": "test"
                    }
                }
            ],
            "16": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::acc"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nacc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 18,
                        "columnEnd": 21,
                        "lineBegin": 16,
                        "lineEnd": 16
                    },
                    "sym": "test/rs/lsif/basics::acc"
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::f1_go"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nbasics\n```\n\n```rust\nfn f1_go(n: u64, acc: u64) -> u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 4,
                        "columnEnd": 9,
                        "lineBegin": 16,
                        "lineEnd": 16
                    },
                    "sym": "test/rs/lsif/basics::f1_go"
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 10,
                        "columnEnd": 11,
                        "lineBegin": 16,
                        "lineEnd": 16
                    },
                    "sym": "test/rs/lsif/basics::n"
                }
            ],
            "17": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::acc"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nacc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 17,
                        "columnEnd": 20,
                        "lineBegin": 17,
                        "lineEnd": 17
                    },
                    "sym": "test/rs/lsif/basics::acc",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 18,
                            "columnEnd": 21,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::acc"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nacc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 43,
                        "columnEnd": 46,
                        "lineBegin": 17,
                        "lineEnd": 17
                    },
                    "sym": "test/rs/lsif/basics::acc",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 18,
                            "columnEnd": 21,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::f1_go"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nbasics\n```\n\n```rust\nfn f1_go(n: u64, acc: u64) -> u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 30,
                        "columnEnd": 35,
                        "lineBegin": 17,
                        "lineEnd": 17
                    },
                    "sym": "test/rs/lsif/basics::f1_go",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 4,
                            "columnEnd": 9,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 8,
                        "columnEnd": 9,
                        "lineBegin": 17,
                        "lineEnd": 17
                    },
                    "sym": "test/rs/lsif/basics::n",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 10,
                            "columnEnd": 11,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 36,
                        "columnEnd": 37,
                        "lineBegin": 17,
                        "lineEnd": 17
                    },
                    "sym": "test/rs/lsif/basics::n",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 10,
                            "columnEnd": 11,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 49,
                        "columnEnd": 50,
                        "lineBegin": 17,
                        "lineEnd": 17
                    },
                    "sym": "test/rs/lsif/basics::n",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 10,
                            "columnEnd": 11,
                            "lineBegin": 16,
                            "lineEnd": 16
                        },
                        "repository": "test"
                    }
                }
            ],
            "22": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 15,
                        "columnEnd": 16,
                        "lineBegin": 22,
                        "lineEnd": 22
                    },
                    "sym": "test/rs/lsif/basics::n"
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::sum_f2"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nbasics\n```\n\n```rust\npub fn sum_f2(n: u64) -> u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 8,
                        "columnEnd": 14,
                        "lineBegin": 22,
                        "lineEnd": 22
                    },
                    "sym": "test/rs/lsif/basics::sum_f2"
                }
            ],
            "23": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut acc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 13,
                        "columnEnd": 16,
                        "lineBegin": 23,
                        "lineEnd": 23
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/23/13/23/16"
                }
            ],
            "24": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 17,
                        "columnEnd": 18,
                        "lineBegin": 24,
                        "lineEnd": 24
                    },
                    "sym": "test/rs/lsif/basics::n",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 15,
                            "columnEnd": 16,
                            "lineBegin": 22,
                            "lineEnd": 22
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut m: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 13,
                        "columnEnd": 14,
                        "lineBegin": 24,
                        "lineEnd": 24
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/24/13/24/14"
                }
            ],
            "26": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut m: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 12,
                        "columnEnd": 13,
                        "lineBegin": 26,
                        "lineEnd": 26
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/24/13/24/14",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 14,
                            "lineBegin": 24,
                            "lineEnd": 24
                        },
                        "repository": "test"
                    }
                }
            ],
            "27": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut acc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 20,
                        "columnEnd": 23,
                        "lineBegin": 27,
                        "lineEnd": 27
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/23/13/23/16",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 16,
                            "lineBegin": 23,
                            "lineEnd": 23
                        },
                        "repository": "test"
                    }
                }
            ],
            "29": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut acc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 13,
                        "columnEnd": 16,
                        "lineBegin": 29,
                        "lineEnd": 29
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/23/13/23/16",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 16,
                            "lineBegin": 23,
                            "lineEnd": 23
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut m: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 20,
                        "columnEnd": 21,
                        "lineBegin": 29,
                        "lineEnd": 29
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/24/13/24/14",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 14,
                            "lineBegin": 24,
                            "lineEnd": 24
                        },
                        "repository": "test"
                    }
                }
            ],
            "30": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut m: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 13,
                        "columnEnd": 14,
                        "lineBegin": 30,
                        "lineEnd": 30
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/24/13/24/14",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 14,
                            "lineBegin": 24,
                            "lineEnd": 24
                        },
                        "repository": "test"
                    }
                }
            ],
            "37": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 15,
                        "columnEnd": 16,
                        "lineBegin": 37,
                        "lineEnd": 37
                    },
                    "sym": "test/rs/lsif/basics::n"
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::sum_f3"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nbasics\n```\n\n```rust\npub fn sum_f3(n: u64) -> u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 8,
                        "columnEnd": 14,
                        "lineBegin": 37,
                        "lineEnd": 37
                    },
                    "sym": "test/rs/lsif/basics::sum_f3"
                }
            ],
            "38": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut acc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 13,
                        "columnEnd": 16,
                        "lineBegin": 38,
                        "lineEnd": 38
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/38/13/38/16"
                }
            ],
            "39": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": "<unknown>"
                        },
                        "symbolParent": {
                            "aString": "basics::n"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nn: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 18,
                        "columnEnd": 19,
                        "lineBegin": 39,
                        "lineEnd": 39
                    },
                    "sym": "test/rs/lsif/basics::n",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 15,
                            "columnEnd": 16,
                            "lineBegin": 37,
                            "lineEnd": 37
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nm: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 9,
                        "columnEnd": 10,
                        "lineBegin": 39,
                        "lineEnd": 39
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/39/9/39/10"
                }
            ],
            "40": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut acc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 9,
                        "columnEnd": 12,
                        "lineBegin": 40,
                        "lineEnd": 40
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/38/13/38/16",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 16,
                            "lineBegin": 38,
                            "lineEnd": 38
                        },
                        "repository": "test"
                    }
                },
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nm: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 16,
                        "columnEnd": 17,
                        "lineBegin": 40,
                        "lineEnd": 40
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/39/9/39/10",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 9,
                            "columnEnd": 10,
                            "lineBegin": 39,
                            "lineEnd": 39
                        },
                        "repository": "test"
                    }
                }
            ],
            "42": [
                {
                    "attributes": {
                        "symbolLanguage": {
                            "aInteger": 10
                        },
                        "symbolName": {
                            "aString": ""
                        },
                        "symbolParent": {
                            "aString": "<unknown>"
                        },
                        "symbolSignature": {
                            "aString": "\n```rust\nlet mut acc: u64\n```"
                        }
                    },
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 8,
                        "lineBegin": 42,
                        "lineEnd": 42
                    },
                    "sym": "test/rs/src/lib.rs/%3Cglobal%3E/38/13/38/16",
                    "target": {
                        "filepath": "src/lib.rs",
                        "range": {
                            "columnBegin": 13,
                            "columnEnd": 16,
                            "lineBegin": 38,
                            "lineEnd": 38
                        },
                        "repository": "test"
                    }
                }
            ]
        },
        "truncated": false
    }
]